### PR TITLE
Improve Documentation for Non-CellProfiler Datasets in Pycytominer

### PR DESCRIPTION
**Pull request by axiomcura**
# Description

Thank you for your contribution to pycytominer!
Please _succinctly_ summarize your proposed change.
What motivated you to make this change?

This PR updates the README with instructions on how to handle morphological features extracted from software other than CellProfiler. These changes were conducted due to a reviewer’s comment/suggestion.

>"When I used the function pycytominer.normalize() on a dataset originating from my own work, I encountered an error message saying
“No CP features found. Are you sure this dataframe is from CellProfiler?”. 
I could easily fix the problem because I know the properties of a CellProfiler dataset, and the error message was hinting at some “Metadata_” missing. 
The fix was very easy, I just needed to add the prefix “Metadata_” to my metadata columns. My dataset comes from another commonly used proprietary software called Harmony. I understand that the authors are in favor of using open-source resources, but I think that the pycytominer tool could be useful in a much larger context, and take as input different types of datasets. In the readme file, that I followed to perform the normalization, I couldn’t find anywhere that the dataset has to come from CellProfiler. The author should state clearly, that only cellprofiler datasets can be used. Instead, currently, the sentence in the readme files of the pycytominer github repository says “Image data flow from a microscope to cell segmentation and feature extraction tools (e.g. CellProfiler or DeepProfiler)”. And in the code chunck the sentence says “Each function takes as input a pandas DataFrame or file path”, but in reality, it has to be a specifically formatted pandas dataframe (at least regarding the column names). I think the authors should fix this misunderstanding in the documentation. Alternatively, the better option would be to support the use of different datasets, and explain how the column names should be formatted to be easily processed by the different functions. I can’t exclude that this is described somewhere in the tutorials, but I didn’t encounter it when following the instructions on the readmefile. I would also make the error message more self-explanatory because for now, it implies that I know what CellProfiler is and how its output is formatted. If I don’t have this knowledge, as a user, I might give up on using this tool. This would be unfortunate, as I believe that this tool is a fundamental step in image-based profiling. Considering how easy was to fix this issue for my own dataset, which allowed me to immediately see the outputs of your tool, I recommend reconsidering how to best document the formatting steps necessary to be able to use the different functions of the pycytominer too"

Please also link to any relevant issues that your code is associated with.

## What is the nature of your change?

- [ ] Bug fix (fixes an issue).
- [ ] Enhancement (adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [X] This change requires a documentation update.

# Checklist

Please ensure that all boxes are checked before indicating that a pull request is ready for review.

- [X] I have read the [CONTRIBUTING.md](CONTRIBUTING.md) guidelines.
- [X] My code follows the style guidelines of this project.
- [X] I have performed a self-review of my own code.
- [X] I have commented my code, particularly in hard-to-understand areas.
- [X] I have made corresponding changes to the documentation.
- [X] My changes generate no new warnings.
- [X] New and existing unit tests pass locally with my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have deleted all non-relevant text in this pull request template.

### CODE REVIEW

File: README.md
Comment (gwaybio) [inline README.md:171]:
```suggestion
```

File: README.md
Comment (gwaybio) [inline README.md:81]:
```suggestion
These features follow a specific structure.
```

File: README.md
Comment (gwaybio) [inline README.md:140]:
```suggestion
Pycytominer also supports processing of raw morphological features from image analysis tools beyond [CellProfiler](https://cellprofiler.org/).
These tools include In Carta, Harmony, and others.
Using Pycytominer with these tools requires minor modifications to function arguments, and we encourage these users to pay particularly close attention to individual function documentation.

If you are using Pycytominer with these other tools, we'd love to hear from you so that we can learn how to best support broad and multiple use-cases.
```

File: README.md
Comment (gwaybio) [inline README.md:142]:
```suggestion
For example, to resolve potential feature issues in the `normalize()` function, you can manually specify the morphological features using the `features` [parameter](https://pycytominer.readthedocs.io/en/latest/pycytominer.html#pycytominer.normalize.normalize).
```

File: README.md
Comment (gwaybio) [inline README.md:143]:
```suggestion
This parameter is also available in other key steps, such as [`aggregate`](https://pycytominer.readthedocs.io/en/latest/pycytominer.html#pycytominer.aggregate.aggregate) and [`feature_select`](https://pycytominer.readthedocs.io/en/latest/pycytominer.html#pycytominer.feature_select.feature_select).
```

File: README.md
Comment (gwaybio) [inline README.md:148]: @axiomcura - maybe the reason why this is marked as a draft, but this example _is_ using a CellProfiler output

Comment (axiomcura) [reply README.md:151]: You’re right, the code and link provided does point to CellProfiler. I was considering adding a comment like ‘assuming this dataset was not generated by CellProfiler,’ but in hindsight, I think this might still end up confusing users.

Are there any datasets available that weren’t generated by CellProfiler? Unfortunately, the InCarta data is private, so we can’t use it as an example. 

I can also remove those links and add placeholders like ‘link to non-CellProfiler data,’ which we can update once we have an appropriate link available.

**Comment by gwaybio** [comment]
@axiomcura - please take a look at my changes. They aim to simplify and clarity. I also removed the non-CellProfiler example that actually used CellProfiler data. This is too confusing and I think the updated function docstrings and error messages in #448 are sufficient. Lastly, I removed the table of contents which was in a strange place and I felt added more confusion than help

**Review by kenibrewer — APPROVED** [review]
Good set of changes! I like adding a description for using these other feature sets front and center.

**Comment by axiomcura** [comment]
I like the changes @gwaybio! I agree that having the table of contents in the middle of the README was a bit confusin. It's much cleaner and more readable now.

@d33bs, waiting on your approval if the updated documentation is ready to be merged! c:

File: README.md
Comment (d33bs) [inline README.md:71]: Consider removing the single quotes here (it didn't seem like there was a need to include these).

```suggestion
Currently, Pycytominer fully supports data generated by [CellProfiler](https://cellprofiler.org/), adhering defaults to its specific data structure and naming conventions.
```

File: README.md
Comment (d33bs) [inline README.md:76]: Is there an example of a non-default compartment name we could show here?

Comment (d33bs) [inline README.md:76]: Are the defaults mentioned here defaults from Pycytominer or CellProfiler's perspective? Could it be important to signify which somehow?

File: README.md
Comment (d33bs) [inline README.md:78]: Minor rewording suggestion.

```suggestion
Note, [`pycytominer.cyto_utils.cells.SingleCells()`](pycytominer/cyto_utils/cells.py) contains code designed to interact with single-cell SQLite files exported from CellProfiler.
```

File: README.md
Comment (d33bs) [inline README.md:79]: Minor suggestion to be specific about what type of cores (I don't believe GPU are involved).

```suggestion
Processing capabilities for SQLite files depends on SQLite file size and your available computational resources (for ex. memory and CPU).
```

File: README.md
Comment (d33bs) [inline README.md:84]: Suggestion to include links to these products, in alignment with what we do for CellProfiler.
```suggestion
These tools include [In Carta](https://www.moleculardevices.com/products/cellular-imaging-systems/high-content-analysis/in-carta-image-analysis-software), [Harmony](https://www.revvity.com/product/harmony-5-2-office-revvity-hh17000019#product-overview), and others.
```

File: README.md
Comment (d33bs) [inline README.md:88]:
```suggestion
The `features` parameter is also available in other key steps, such as [`aggregate`](https://pycytominer.readthedocs.io/en/latest/pycytominer.html#pycytominer.aggregate.aggregate) and [`feature_select`](https://pycytominer.readthedocs.io/en/latest/pycytominer.html#pycytominer.feature_select.feature_select).
```

File: README.md
Comment (d33bs) [inline README.md:90]: Should we indicate "how" to hear from people here (issues?). Aside, could it be useful to enable GitHub Discussions to help with community engagement on this front?

**Review by d33bs — APPROVED** [review]
Nice job! I left a few comments + suggestions and overall thought this LGTM when ready. Additionally, had these thoughts:

- I noticed that the `Citing pycytominer` header might need to be capitalized to stay consistent with other uses in the document.
- I noticed the image of the workflows could use maybe another block alongside CellProfiler (or some other method) to indicate the freedom to use other tools as desired (at the moment it visually communicates you can _only_ use two tools to reach the workflows). Maybe there could be a mention of `"Single-cell Profiling Software (e.g. CellProfiler, DeepProfiler, etc.)"` (this is a bit too long but I think you can see what I mean!). See below:

image

**Comment by gwaybio** [comment]
It looks like this is a duplicate of #448 ? @axiomcura - is this true? How are these different? 

Taking a look at what's changed here, it looks like #448 is the correct PR and we should close this one.

**Comment by gwaybio** [comment]
(I'll plan on going through @d33bs comments and adding them to 448)

**Comment by gwaybio** [comment]
oh wait! I am the one confused. One sec

File: README.md
Comment (axiomcura) [reply README.md:76]: I don't have experience working with non-default settings, but one potential issue could be changes in compartment names, such as differences in capitalization. In such cases, users would need to use the compartments argument as shown [here](https://github.com/cytomining/pycytominer/blob/4b772a5d7e7ded6c5ae67a024f28102c4d2305db/pycytominer/cyto_utils/util.py#L18). This adjustment would also affect the linking of columns as well [here](https://github.com/cytomining/pycytominer/blob/4b772a5d7e7ded6c5ae67a024f28102c4d2305db/pycytominer/cyto_utils/single_cell_ingest_utils.py#L5).

Comment (gwaybio) [reply README.md:76]: i think @d33bs is just asking for simple example, which I like the idea of. I'll make the change

Comment (gwaybio) [reply README.md:76]: I think it is ok to leave as is

File: README.md
Comment (gwaybio) [reply README.md:90]: great point! I will address

**Comment by gwaybio** [comment]
Thanks for the review @d33bs !

One thing I left for @axiomcura to address is this comment:

> I noticed the image of the workflows could use maybe another block alongside CellProfiler (or some other method) to indicate the freedom to use other tools as desired (at the moment it visually communicates you can only use two tools to reach the workflows). Maybe there could be a mention of "Single-cell Profiling Software (e.g. CellProfiler, DeepProfiler, etc.)" (this is a bit too long but I think you can see what I mean!). See below:

@axiomcura - what do you think about dropping this figure in favor of your new figure 1? I propose that you rename the existing file (maybe to something like `pipeline_legacy.png`) and add a high res version of figure one (naming it `pipeline.png`)

File: README.md
Comment (gwaybio) [inline README.md:18]: again, I think we need full for RTD

File: README.md
Comment (gwaybio) [inline README.md:25]:
```suggestion
Next, Pycytominer performs reproducible image-based profiling (**Figure 1B**).
The Pycytominer API consists of five key steps (**Figure 1C**).
```

File: README.md
Comment (gwaybio) [inline README.md:20]:
```suggestion
Image data flow from a microscope to cell segmentation and feature extraction tools (e.g. [CellProfiler](https://cellprofiler.org/) or [DeepProfiler](https://cytomining.github.io/DeepProfiler-handbook/docs/00-welcome.html)) (**Figure 1A**).
```

File: README.md
Comment (gwaybio) [inline README.md:26]:
```suggestion
The outputs generated by Pycytominer are utilized for downstream analysis, which includes machine learning models and statistical testing to derive biological insights.
```

**Review by gwaybio — CHANGES_REQUESTED** [review]

**Review by gwaybio — APPROVED** [review]
LGTM! 🎉 - I'll merge this in once tests pass

**Comment by gwaybio** [comment]
update, looks like macos runners aren't picking up the jobs. Since ubuntu and other integration tests passed and this PR doesn't touch code, I will merge to reduce delays

**Pull request merged**
